### DISCUSSION TITLE
Create superUser

### DISCUSSION THREAD

**Discussion by rithikyennam** [nautobot/nautobot#3436] [Q&A] [answered]
Is there a way to create a superuser using Nautobot API
I see there is no is_superuser flag in the API for creating users.

Is there any other way of creating nautobot superuser using a python script?

If  python script is not possible can you suggest a way to create it using ansible module?
We are using ansible module for deployment, using ansible it is allowing to create a super user but we cannot set a password for it

**Comment by gsnider2195** [comment, marked as answer]
It looks like the only option right now is to use the [ansible.builtin.expect](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/expect_module.html) module to run `nautobot-server createsuperuser` and answer the prompts. You could use `nautobot-server createsuperuser --noinput --username theusername --email me@example.com` to create the user without a password but the API doesn't currently support updating passwords on existing users, only setting passwords on new users created through the API.

**Reply by jvanderaa** [reply]
@gsnider2195 there is a nautobot-server module that can handle this.

**Reply by jvanderaa** [reply]
@rithikyennam - https://nautobot-ansible.readthedocs.io/en/latest/plugins/nautobot_server_module.html#ansible-collections-networktocode-nautobot-nautobot-server-module

**Comment by gsnider2195** [comment]
Please refer to issues #2208 and #3438 (created from this discussion)
